src/DotLiquid/Util/StrFTime.cs

```diff
@@ -16,87 +21,201 @@ public static class StrFTime
+            { "z", (dateTimeOffset) => dateTimeOffset.ToString("%K", CultureInfo.CurrentCulture).Replace(":", string.Empty) },
+            { ":z", (dateTimeOffset) => dateTimeOffset.ToString("%K", CultureInfo.CurrentCulture) },
             { "Z", (dateTimeOffset) => dateTimeOffset.ToString("zzz", CultureInfo.CurrentCulture) },
         };
 
-        public static string ToStrFTime(this DateTime dateTime, string pattern)
+        /// <summary>
+        /// Applies formatting consistent to the rules specified by the Ruby Time.strftime function.
+        /// The following exceptions apply;
+        /// - `%N` (Not standard) - DateTime does not support nanoseconds precision, so the default is microsecond (6dp).
+        /// - `%9N` (Not implemented) - DateTime does not support nanoseconds precision.
+        /// - `%Z` (Not standard) - DotLiquid returns an offset (+00:00), Ruby states this should return a timezone description (e.g. 'GMT')
+        /// <see href="https://help.shopify.com/themes/liquid/filters/additional-filters#date"/>
+        /// <see href="https://ruby-doc.org/core-3.0.0/Time.html#method-i-strftime"/>
+        /// </summary>
+        /// <param name="dateTime">date-time object to be formatted</param>
+        /// <param name="format">the required format</param>
+        /// <returns>a string version of date-time matching pattern.</returns>
+        public static string ToStrFTime(this DateTime dateTime, string format)
         {
-            string output = "";
+            return Regex.Replace(input: format, pattern: PATTERN_REGEX,
+                evaluator: x => StrFTimeMatchEvaluator(
+                    x.Groups[0].Value,
+                    x.Groups["flag"].Captures.Cast<Capture>().Select(y => y.Value).ToList(),
+                    x.Groups["width"].Captures.Cast<Capture>().Select(y => (int?)Convert.ToInt32(y.Value)).FirstOrDefault(),
+                    x.Groups["directive"].Captures.Cast<Capture>().Select(y => y.Value).FirstOrDefault(),
+                    dateTime
+                    ));
+        }
 
-            int n = 0;
+        /// <summary>
+        /// Formats a date using a ruby date format string
+        /// </summary>
+        /// <param name="dateTimeOffset">date-time object to be formatted</param>
+        /// <param name="format">the required format</param>
+        /// <returns>a string version of date-time matching pattern.</returns>
+        public static string ToStrFTime(this DateTimeOffset dateTimeOffset, string format)
+        {
+            return Regex.Replace(input: format, pattern: PATTERN_REGEX,
+                evaluator: x => StrFTimeMatchEvaluator(
+                    x.Groups[0].Value,
+                    x.Groups["flag"].Captures.Cast<Capture>().Select(y => y.Value).ToList(),
+                    x.Groups["width"].Captures.Cast<Capture>().Select(y => (int?)Convert.ToInt32(y.Value)).FirstOrDefault(),

```

Comment:
rename 'y' to an English word describing what is contained in the variable.

Reply:
Done